homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(hass, session, hivedevice)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hass, Session, HiveDevice):
+        """Initialize the Climate device."""
+        self.node_id = HiveDevice["Hive_NodeID"]
+        self.node_name = HiveDevice["Hive_NodeName"]
+        self.device_type = HiveDevice["HA_DeviceType"]
+        self.hass = hass
+        self.session = Session
+
+        if self.device_type == "Heating":
+            self.session.heating = self.session.core.Heating()
+            self.modes = [STATE_AUTO, STATE_HEAT, STATE_OFF]
+        elif self.device_type == "HotWater":
+            self.session.hotwater = self.session.core.Hotwater()
+            self.modes = [STATE_AUTO, STATE_ON, STATE_OFF]
+
+        self.session.entities.append(self)
+
+    def handle_update(self, updatesource):
+        """Handle the new update request."""
+        if self.device_type + "." + self.node_id not in updatesource:

```

Comment:
Use new style string formatting:
```py
'{}.{}'.format(self.device_type, self.node_id)
```

Reply:
so `if self.device_type + "." + self.node_id not in updatesource:` needs to change to `if '{}.{}'.format(self.device_type, self.node_id) not in updatesource:`?

Reply:
done, I think